### DISCUSSION TITLE
Place poetry exactly at the center of the page

### DISCUSSION THREAD

**Discussion by johnmapeson** [typst/typst#4894] [Q&A] [answered]
Currently, for poetry, I use the following markup:

```
This is prose. Lorem ipsum dolor sit amet, consectetur adipiscing elit.

#h(25%) This \
#h(25%) is \
#h(25%) poetry.

This is prose. Lorem ipsum dolor sit amet, consectetur adipiscing elit.
```

But is it possible to place the poetry block exactly at the middle of the page (horizontally, of course, nothing unusual), instead of using an indent such as `25%` or `1in`?

I tried `#align`, but then each line is centered, whereas what I want is to center the block itself.

```
#align(center)[
  This \
  is \
  poetry.
]
```

**Comment by jamesrswift** [comment, marked as answer]
I'm not at a computer to test whether this answer is correct but have you considered something like `pad(x: 1fr, [...])`?

**Reply by freetz13** [reply]
No longer working in Typst at least v0.13.1: `error: expected relative length, found fraction`

**Comment by johnmapeson** [comment]
Thanks, James. Works for me.

```
a b c d e f g h j k l m n o p q r s t u v w x y z . . . a b c d e f g h j k l m n o p q r s t u v w x y z . . . a b c d e f g h j k l m n o p q r s t u v w x y z

#pad(x: 25%, [
  a b c d e f g h j k l m n o p q r s t u v w x y z . . . a b c d e f g h j k l m n o p q r s t u v w x y z . . . a b c d e f g h j k l m n o p q r s t u v w x y z
])

a b c d e f g h j k l m n o p q r s t u v w x y z . . . a b c d e f g h j k l m n o p q r s t u v w x y z . . . a b c d e f g h j k l m n o p q r s t u v w x y z
```
